tools/src/publish-packages/tasks/publishAndroidPackages.ts

```diff
@@ -0,0 +1,141 @@
+import spawnAsync from '@expo/spawn-async';
+import { green } from 'chalk';
+import fs from 'fs-extra';
+import path from 'path';
+
+import { EXPO_DIR } from '../../Constants';
+import logger from '../../Logger';
+import { ExpoModuleConfig, Package } from '../../Packages';
+import { Task } from '../../TasksRunner';
+import type { CommandOptions, Parcel } from '../types';
+import { updateAndroidProjects } from './updateAndroidProjects';
+
+const EXCLUDE = ['expo-module-template-local', 'expo-module-template'];
+
+export const publishAndroidArtifacts = new Task(
+  {
+    name: 'publishAndroidArtifacts',
+    dependsOn: [updateAndroidProjects],
+    filesToStage: ['packages/**/expo-module.config.json'],
+  },
+  async (parcels: Parcel[], options: CommandOptions) => {
+    const packages = parcels.map((parcels) => parcels.pkg);
+
+    // Collect all packages that have Android artifacts to publish.
+    const androidPackages = packages.filter((pkg) => {
+      const moduleConfig = pkg.expoModuleConfig;
+      if (!moduleConfig) {
+        return false;
+      }
+      if (!moduleConfig.platforms.includes('android')) {
+        return false;
+      }
+      if (EXCLUDE.includes(pkg.packageName)) {
+        return false;
+      }
+      return true;
+    });
+
+    if (androidPackages.length === 0) {
+      logger.log('\n🤖 No Android artifacts to publish.');
+      return;
+    }
+
+    logger.info('\n🤖 Publishing Android artifacts...');
+
+    logger.log('  ', 'Removing existing publications...');
+    // Remove existing publications from expo-module.config.json
+    await Promise.all(androidPackages.map((pkg) => removeExistingPublications(pkg)));
+
+    const batch = logger.batch();
+    batch.log('  ', 'Collecting publication Gradle tasks...');
+    // Collect all Gradle tasks to publish Android artifacts.
+    // We need to use `expoPublishToMavenLocal` for dry run.
+    // Otherwise, we use `expoPublish`.
+    const gradleTasks = androidPackages.flatMap((pkg) => {
+      const publicationCommands = getPublicationCommands(
+        pkg.packageName,
+        pkg.expoModuleConfig!,
+        options
+      );
+
+      if (publicationCommands.length === 1) {
+        const publicationCommand = publicationCommands[0];
+        batch.log('  ', '  ', `${green(pkg.packageName)}: ${publicationCommand}`);
+      } else {
+        batch.log('  ', '  ', `${green(pkg.packageName)}:`);
+        for (const command of publicationCommands) {
+          batch.log('  ', '  ', '  ', `${command}`);
+        }
+      }
+
+      return publicationCommands;
+    });
+    batch.flush();
+
+    logger.log('  ', 'Publishing Android artifacts...');
+    // Run Gradle tasks in one batch to speed up the process.
+    await spawnAsync('./gradlew', gradleTasks, {
```

Comment:
We are running all projects in one project to save time. Otherwise, this process will be very painful.